jenkins/scripts/select-compiler.sh

```diff
@@ -4,14 +4,14 @@ if [ "$DONTSELECT_COMPILER" != "DONT" ]; then
   NODE_NAME=${NODE_NAME:-$HOSTNAME}
   echo "Selecting compiler based on $NODE_NAME"
   case $NODE_NAME in
-    *ppc64_le* ) SELECT_ARCH=PPC64LE ;;
+    *ppc64*le* ) SELECT_ARCH=PPC64LE ;;
```

Comment:
It this because the node names are inconsistent?

Reply:
because of https://github.com/nodejs/build/blob/6d12c28b01dbdaaf71fc810dcd11e5ffc3095d5d/ansible/roles/baselayout/tasks/main.yml#L16, it means that `HOSTNAME := NODE_NAME.replace(_, --)`, so select compiler's attempt to use the HOSTNAME as a default, which usually works, fails here.

Reply:
The change should be OK, but again it's an oddity because `NODE_NAME` should come from Jenkins and the only situation where it's not set is when this is run independently, I guess if this is run as an independent test?

Reply:
Yes, anyone attempting to trouble shoot build issues or bootstrap a ci machine needs to read select-compiler by hand, after reading it to figure out what they should set NODE_NAME to. I added HOSTNAME as the default. If we don't want that, I can back it out. Given that ansible sets the HOSTNAME to the NODE_NAME (modulo the `-` and `_`), I'm not sure why this script depends on ci env vars. If this change is out of scope, I'll back it out. Its more of a public service for future users.

Reply:
Ok, understood now  and makes sense.  The result is that you can call select-compiler.sh on the host by hand and get the right behavior without having to export NODE_NAME.

Reply:
+1